js/src/logger.ts

```diff
@@ -1376,19 +1413,21 @@ export function init<IsOpen extends boolean = false>(
 
     const lazyMetadata: LazyValue<ProjectExperimentMetadata> = new LazyValue(
       async () => {
-        await login({
-          orgName: orgName,
-          apiKey,
-          appUrl,
-        });
+        const state =
+          stateArg ??
+          (await login({
```

Comment:
I kind-of imagined this would look like `(stateArg ?? _dangerousGlobalState).login(loginParams)`. And then the public `login` function is just `_dangerousGlobalState.login(loginParams)`. That way the only global thing any of these codepaths need to access is `_dangerousGlobalState`. The other nice thing is that you could just have one `const state = stateArg ?? _dangerousGlobalState` at the beginning of this function and use that everywhere else.

Is there a reason something like that wouldn't work?

Reply:
oh that's a clever idea. Let me rework.